api/content.go

```diff
@@ -0,0 +1,202 @@
+func (api *API) getContentPublicHandler(w http.ResponseWriter, req *http.Request) {
+	ctx := req.Context()
+	vars := mux.Vars(req)
+	id := vars["id"]
+	logdata := log.Data{
+		"request_id": ctx.Value(dprequest.RequestIdKey),
+		"content_id": id,
+		"function":   "getContentPublicHandler",
+	}
+
+	// check topic from mongoDB by id
+	_, err := api.dataStore.Backend.GetTopic(id)
+	if err != nil {
+		handleError(ctx, w, err, logdata)
+		return
+	}
+
+	// get content from mongoDB by id
+	content, err := api.dataStore.Backend.GetContent(id)
+	if err != nil {
+		// no content found
+		handleError(ctx, w, err, logdata)
+		return
+	}
+
+	// User is not authenticated and hence has only access to current sub document(s)
+
+	if content.Current == nil {
+		handleError(ctx, w, apierrors.ErrInternalServer, logdata)
+		return
+	}
+
+	var currentResult models.ContentResponseAPI
+
+	// Add spotlight first
+	addItem(&currentResult, "spotlight", content.Current.Spotlight, content.ID, content.Current.State, false)
+	// then Publications (alphabetically ordered)
+	addItem(&currentResult, "articles", content.Current.Articles, content.ID, content.Current.State, false)
+	addItem(&currentResult, "bulletins", content.Current.Bulletins, content.ID, content.Current.State, false)
+	addItem(&currentResult, "methodologies", content.Current.Methodologies, content.ID, content.Current.State, false)
+	addItem(&currentResult, "methodologyArticles", content.Current.MethodologyArticles, content.ID, content.Current.State, false)
+	// then Datasets (alphabetically ordered)
+	addItem(&currentResult, "staticDatasets", content.Current.StaticDatasets, content.ID, content.Current.State, false)
+	addItem(&currentResult, "timeseries", content.Current.Timeseries, content.ID, content.Current.State, false)
+
+	currentResult.Count = currentResult.TotalCount // This may be '0' which is the case for some existing ONS pages (like: bankruptcyinsolvency as of 3.feb.2021)
+
+	if err := WriteJSONBody(ctx, currentResult, w, logdata); err != nil {
+		return
+	}
+	log.Event(ctx, "request successful", log.INFO, logdata) // NOTE: name of function is in logdata
+}
+
+// getContentPrivateHandler is a handler that gets content by its id from MongoDB for Publishing
+func (api *API) getContentPrivateHandler(w http.ResponseWriter, req *http.Request) {
+	ctx := req.Context()
+	vars := mux.Vars(req)
+	id := vars["id"]
+	logdata := log.Data{
+		"request_id": ctx.Value(dprequest.RequestIdKey),
+		"content_id": id,
+		"function":   "getContentPrivateHandler",
+	}
+
+	// check topic from mongoDB by id
+	_, err := api.dataStore.Backend.GetTopic(id)
+	if err != nil {
+		handleError(ctx, w, err, logdata)
+		return
+	}
+
+	// get content from mongoDB by id
+	content, err := api.dataStore.Backend.GetContent(id)
+	if err != nil {
+		// no content found
+		handleError(ctx, w, err, logdata)
+		return
+	}
+
+	// User has valid authentication to get raw full content document(s)
+
+	if content.Current == nil {
```

Comment:
When a document is first created, it will only have 'next' until it is published, when it gets 'current' populated. So current == nil is not an error. If for now we want to return an error because we dont have publishing steps, then this if statement should include a comment explaining that this behaviour is expected to change when the API becomes more than read-only

Reply:
Comment along the lines of the above added.